flex-backend/api/http/server.go

```diff
@@ -15,50 +15,76 @@ type Server struct {
 }
 
 func (s Server) BuildEndpoints() {
-	http.HandleFunc("/getMovies", handleGetMovies)
-	http.HandleFunc("/getInfo", handleGetMovieInfo)
+	http.HandleFunc("/getMovies", s.handleGetMovies)
+	http.HandleFunc("/getInfo", s.handleGetMovieInfo)
 	http.HandleFunc("/playFile", handlePlayFile)
 	http.HandleFunc("/stopFile", handleStopFile)
 	http.HandleFunc("/ws", s.handleWebSocket)
 	// Used to test the websocket
 	http.HandleFunc("/", handleSendRoot)
 }
 
-func (Server) Serve() {
+func (Server) Serve(port int) {
 	// Start the server
-	log.Println("Starting server on port 8080")
-	log.Fatal(http.ListenAndServe(":8080", nil))
+	log.Printf("Starting server on port %d\n", port)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", port), nil))
 }
 
 func handleSendRoot(w http.ResponseWriter, r *http.Request) {
-	// list out local movies
-	fmt.Println("Called route!")
 	http.ServeFile(w, r, "websockets.html")
 }
 
-func handleGetMovies(w http.ResponseWriter, r *http.Request) {
-	// list out local movies
-	fmt.Fprintf(w, "Hello, %q", html.EscapeString(r.URL.Path))
-	fmt.Println("Called get movies route!")
+func (s Server) handleGetMovies(w http.ResponseWriter, r *http.Request) {
+	// Send back the filepaths to client as json
+	files, err := s.MovieHandler.ReadLocalDir(s.MovieHandler.MovieDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+	jData, err := json.Marshal(files)
+	if err != nil {
```

Comment:
if the json fails to marshal we would want to send back a negative status code instead of just sending them nil. :)

Reply:
This is a good point. I'll go look at it